gpytorch/kernels/kernel.py

```diff
@@ -23,44 +23,51 @@
 from ..priors import Prior
 
 
-def default_postprocess_script(x):
-    return x
-
-
+def _sq_dist(x1, x2, x1_eq_x2=False):
+    # TODO: use torch squared cdist once implemented: https://github.com/pytorch/pytorch/pull/25799
+    adjustment = x1.mean(-2, keepdim=True)
+    x1 = x1 - adjustment
+    x2 = x2 - adjustment  # x1 and x2 should be identical in all dims except -2 at this point
+
+    # Compute squared distance matrix using quadratic expansion
+    x1_norm = x1.pow(2).sum(dim=-1, keepdim=True)
+    x1_pad = torch.ones_like(x1_norm)
+    if x1_eq_x2 and not x1.requires_grad and not x2.requires_grad:
+        x2_norm, x2_pad = x1_norm, x1_pad
+    else:
+        x2_norm = x2.pow(2).sum(dim=-1, keepdim=True)
+        x2_pad = torch.ones_like(x2_norm)
+    x1_ = torch.cat([-2.0 * x1, x1_norm, x1_pad], dim=-1)
+    x2_ = torch.cat([x2, x2_pad, x2_norm], dim=-1)
+    res = x1_.matmul(x2_.transpose(-2, -1))
+
+    if x1_eq_x2 and not x1.requires_grad and not x2.requires_grad:
+        res.diagonal(dim1=-2, dim2=-1).fill_(0)
+
+    # Zero out negative values
+    return res.clamp_min_(0)
+
+
+def _dist(x1, x2, x1_eq_x2=False):
+    # TODO: use torch cdist once implementation is improved: https://github.com/pytorch/pytorch/pull/25799
+    res = _sq_dist(x1, x2, x1_eq_x2=x1_eq_x2)
+    return res.clamp_min_(1e-30).sqrt_()
+
+
+# only necessary for legacy purposes
 class Distance(torch.nn.Module):
-    def __init__(self, postprocess_script=default_postprocess_script):
+    def __init__(self, postprocess: Optional = None):
         super().__init__()
-        self._postprocess = postprocess_script
-
-    def _sq_dist(self, x1, x2, postprocess, x1_eq_x2=False):
-        # TODO: use torch squared cdist once implemented: https://github.com/pytorch/pytorch/pull/25799
-        adjustment = x1.mean(-2, keepdim=True)
-        x1 = x1 - adjustment
-        x2 = x2 - adjustment  # x1 and x2 should be identical in all dims except -2 at this point
-
-        # Compute squared distance matrix using quadratic expansion
-        x1_norm = x1.pow(2).sum(dim=-1, keepdim=True)
-        x1_pad = torch.ones_like(x1_norm)
-        if x1_eq_x2 and not x1.requires_grad and not x2.requires_grad:
-            x2_norm, x2_pad = x1_norm, x1_pad
-        else:
-            x2_norm = x2.pow(2).sum(dim=-1, keepdim=True)
-            x2_pad = torch.ones_like(x2_norm)
-        x1_ = torch.cat([-2.0 * x1, x1_norm, x1_pad], dim=-1)
-        x2_ = torch.cat([x2, x2_pad, x2_norm], dim=-1)
-        res = x1_.matmul(x2_.transpose(-2, -1))
+        if postprocess is not None:
+            warnings.warn("The `postprocess` argument will be deprecated.", DeprecationWarning)
```

Comment:
:)
```suggestion
            warnings.warn("The `postprocess` argument is deprecated.", DeprecationWarning)
```

Reply:
Could add a link to this PR here for people to understand why.

Reply:
Done.